### DISCUSSION TITLE
Should a dry run indicate which tests are selected?

### DISCUSSION THREAD

**Discussion by BeeGrech** [teemtee/tmt#1123] [Q&A]
for example tmt run -a -n plan -n core
where run -n is dry
in the information out- summary displays 0 tests selected
```/plans/features/core
    discover
        how: fmf
        directory: /home/bgrech/tmt/tmt
        filter: tier: 0, 1
        summary: 0 tests selected
    provision
        how: local
        summary: 1 guest provisioned
    prepare
        summary: 0 preparations applied
    execute
        how: tmt
        summary: 0 tests executed
    report
        how: display
        summary: no results found
    finish
        summary: 0 tasks completed
```
where there are 16 tests which would be selected

**Comment by psss** [comment]
In order to show which tests would be run the following command (documented in the [Select Steps](https://tmt.readthedocs.io/en/latest/examples.html#select-steps) section) can be used:

    tmt run discover -v

In dry mode we're trying not to do any real actions, e.g. install packages, fetch repositories or even copy files. Thus it's not possible to fully explore which tests would be run.

Perhaps, it might make sense to make the dry mode working in `discover` at least when local fmf discovery is performed? Not sure. @FrNecas, @lukaszachy, @thrix, @dkarpele what do you think?

**Reply by thrix** [reply]
I believe the documentation needs an updated to be more clear about this. If I read it now:

```
Run in dry mode, just let me know what would be done
```

I would expect discover to happen :) Let's make the use case clear, what is it good for? Is it about informing which steps would be run? If so, maybe we should also omit the `summary` it is clearly misleading the poor user. WDYT @psss @FrNecas @lukaszachy  @dkarpele  ?

**Reply by psss** [reply]
What do you mean by "omit the summary"? Any better suggestions how to describe the `dry` mode?

**Reply by BeeGrech** [reply]
In this case I assumed since the  -a parameter was provided all steps would have been performed including identifying the tests

**Reply by lukaszachy** [reply]
> What do you mean by "omit the summary"?

I believe to don't print `summary: 0 tests selected`, it is misleading

**Reply by psss** [reply]
Understood, makes sense.
